### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by kphutiyagae**
* Converted JS files to TS.
* Added Interfaces.
* Merged Develop into Typescript feature branch.

### CODE REVIEW

File: src/scripts/api-observables.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { Observable } from 'rxjs';
+
+const flightDataObservable$ = new Observable((subscriber) => subscriber.next());
```
Comment (sechaba-mache) [inline src/scripts/api-observables.ts]: No type specified for flightDataObservable$

File: src/scripts/map-service.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+function moveToFlight(latitude: number, longtitude: number): void {
+    if (!latitude || !longtitude) return;
+    map.invalidateSize();
+    map.flyTo([latitude, longtitude], 10, {
+        animate: true,
+    });
+
+}
+
+const mapObserver = {
```
Comment (sechaba-mache) [inline src/scripts/map-service.ts]: Add typing for mapObserver

File: src/scripts/observables.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    catchError((error) => {
+        return of({ error: true, message: `Error ${error.status}` });
+    })
+);
+
+const isRequestInCacheTest: boolean = await isRequestInCache(
+    `${baseUrl}/states/all?extended=1`
+)
+
+const cachedOrRequestedFlightData$ = iif(
```
Comment (sechaba-mache) [inline src/scripts/observables.ts]: Add typing

File: src/types/interfaces.d.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    squak: string;
+    spi: boolean;
+    position_source: number;
+    category: number;
+}
+export interface IFlightArray {
+    flightsArray: IFlight[];
+}
+
+export interface ofError {
```
Comment (sechaba-mache) [inline src/types/interfaces.d.ts]: Bad naming on interface. Start with capital I then name

**Review by sechaba-mache — CHANGES_REQUESTED** [review]
Add whole dist folder to gitignore

File: src/types/interfaces.d.ts (hunk 4 above)
Comment (kphutiyagae) [reply src/types/interfaces.d.ts]: Resolved.

**Review by sechaba-mache — APPROVED** [review]

**Comment by kphutiyagae** [comment]
* Changed Javascript implementation to Typescript.

**Pull request merged**